File: Examen/Ej03.cpp
//Incluimos la clase.
#include "Class_ABB_03.hpp"

//Programa principal.
int main()
{
    //Definimos el arbol.
    ArbolBusquedaBinario *arbol = new ArbolBusquedaBinario();

    string arr[12] = {"G","B","Q","A","C","K","F","P","D","E","R","H"};

    //Agregamos elementos al arbol.
    for (int i = 0; i < 12; i++)
    {
        arbol->agrega(i, arr[i]); ////////Va dar resultados distintos con respecto al de python ya que va a queda un arbol muy pesado a un lado ya que las claves van aumentando de uno en uno.
    }

    cout << "Número de nodos en el árbol: " << arbol->nodos() << endl;
    cout << "Número de hojas en el árbol: " << arbol->hojas() << endl;
    cout << "Profundidad del árbol: " << arbol->profundidad() << endl;

    //Finalizamos el programa.
    cout << "\nPulsa ENTER para finalizar.";
    cin.ignore();
    return 0;
}

File: Examen/Ej03.py
from Class_ArbolesBinariosBusqueda_03 import ArbolBinarioBusqueda


arbol = ArbolBinarioBusqueda()

lista = ["G","B","Q","A","C","K","F","P","D","E","R","H"]

for i in lista:
    arbol.agregar(i, i)

print("a)", end=" ");arbol.preorden()
print("b)", end=" ");arbol.inorden()
print("c)", end=" ");arbol.postorden()
print("d)", end=" ");arbol.anchura()

print("Cantidad de nodos en el árbol:",arbol.nodos())
print("Cantidad de nodos hoja en el árbol:",arbol.hojas())
print("Profundidad del árbol:",arbol.profundidad())